src/saveload/saveload.cpp

```diff
@@ -2336,6 +2449,17 @@ static const SaveLoadFormat _saveload_formats[] = {
 #else
 	{"zlib",   TO_BE32X('OTTZ'), nullptr,                            nullptr,                            0, 0, 0},
 #endif
+#if defined(WITH_ZSTD)
+	/* Zstd provides a decent compression rate at a very high compression/decompression speed. Compared to lzma level 2
+	 * zstd saves are about 40% larger (on level 1) but it has about 30x faster compression and 5x decompression making it
+	 * a good choice for multiplayer servers. And zstd level 1 seems to be the optimal one for client connection speed
+	 * (compress + 10 MB/s download + decompress time), about 3x faster than lzma:2 and 1.5x than zlib:2 and lzo.
+	 * As zstd has negative compression levels the values were increased by 100 moving zstd level range -100..22 into
+	 * openttd 0..122. Also note that value 100 mathes zstd level 0 which is a special value for default level 3 (openttd 103) */
+	{"zstd",   TO_BE32X('OTTS'), CreateLoadFilter<ZSTDLoadFilter>,   CreateSaveFilter<ZSTDSaveFilter>,   0, 101, 122},
+#else
+	{"zstd",   TO_BE32X('OTTS'), nullptr,                            nullptr,                            0, 0, 0},
+#endif
```

Comment:
If I remember correctly, putting it here doesn't make it a default for anyone. Would it be wise to add code that depending on network-save / local-save it picks zstd? As just adding it for the sake of adding it feels a bit mute to me. Opinions? Ideas?

Reply:
Well, my idea was to just add it first and decide what to do with it later (1.12?) so there is enough time for it to be tested by those who wish to.

Reply:
Personally not a back fan of using upstream for this, but that is just my take in this :) Mainly because I also think that adding it isn't a lot of work, and would benefit many people :) (in the end, we know it is faster for sure, and sizes on network games do not matter that much after all). But again, just my take on it, curious if others agree/disagree :)